### PR TITLE
Feature/queries refactoring

### PR DESCRIPTION
**Pull request by jmzwar**
This PR removes the queries library creating an instance of synthetixjs, and replaces it with an instance passed through createQueryContext. Also added error handling and removed many non-null assertions.

### CODE REVIEW

File: packages/queries/src/queries/debt/useGetDebtL1.ts [hunk 1]
```diff
@@ -14,15 +14,18 @@ import {
 } from '../../../generated/mainSubgraphQueries';
 import { DEFAULT_SUBGRAPH_ENDPOINTS } from '../../constants';
 import { Debt, DebtData } from './useGetDebtL2';
-import synthetix, { NetworkIdByName } from '@synthetixio/contracts-interface';
+import { Contract } from '@ethersproject/contracts';
+import { address, abi } from '@synthetixio/contracts/build/mainnet/deployment/ExchangeRates';
```
[the remark this answers is not in the document]
Comment (jmzwar) [reply packages/queries/src/queries/debt/useGetDebtL1.ts:18]: This is useGetDebtL1, there is also a useGetDebtL2. We could make it dynamic but didn't want to do that in this pr.

File: packages/queries/src/index.ts [hunk 2]
```diff
@@ -79,12 +81,11 @@ export function createQueryContext({
     subgraphEndpoints: subgraphEndpoints || DEFAULT_SUBGRAPH_ENDPOINTS[1],
   };
 
-  if (networkId) {
-    ctx.snxjs = synthetix({ networkId, signer, provider });
+  if (networkId && synthetixjs && signer && provider) {
+    ctx.snxjs = synthetixjs;
```
Comment (bachstatter) [inline packages/queries/src/index.ts:85]: Is instantiating synthetixjs costly performance wise?  It's a fairly breaking change. We probably need to do a major bump? And talk to kwenta.

Maybe we should still instantiate if it passed in as null?

**Review by bachstatter — APPROVED** [review]
Makes sense to me. It's more a question how backwards compatible we want to be

File: packages/queries/src/queries/rates/useSynthMarketCapQuery.ts [hunk 3]
```diff
@@ -14,7 +14,8 @@ const useSynthMarketCapQuery = (
   return useQuery<Wei>(
     ['rates', 'marketCap', ctx.networkId, currencyKey],
     async () => {
-      return wei(await ctx.snxjs!.contracts[synthToContractName(currencyKey!)].totalSupply());
+      if (!ctx.snxjs) throw Error('Expected snxjs to be defined');
```
[the remark this answers is not in the document]
Comment (bachstatter) [reply packages/queries/src/queries/rates/useSynthMarketCapQuery.ts:17]: Maybe you talked about it, but even if it is added to the enabled option we need to trow to "help" typescript no that it's not undefined

**Pull request merged**